### PR TITLE
Use `urllib3.Retry` instead of custom implementation

### PR DESCRIPTION
**Pull request by crazyscientist**
* Removed custom implementation
* Moved session-related code into an utility module
* Removed request logging

### CODE REVIEW

File: osctiny/utils/session.py [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    }
+
+    # Older versions of Retry do not understand some parameters:
+    if version < (1, 26, 0):
+        kwargs.update({
+            "method_whitelist": None,
+        })
+    else:
+        new_kwargs = {
+            "allowed_methods": None,  # Aka. allow all!
```
Comment (der-gabe) [inline osctiny/utils/session.py:60]: I hate that `None` means "all" in `urllib3.util.Retry`, but that's how they decided to implement it...

Thanks for documenting it! :slightly_smiling_face: :+1:

File: osctiny/utils/session.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+def generate_session_id(username: str, url: str) -> str:
+    """
+    Generate a session ID unique to the user, remote host, process and thread.
+    """
+    session_hash = b64encode(f'{username}@{url}'.encode()).decode()
+    return f"session_{session_hash}_{os.getpid()}_{threading.get_ident()}"
+
+
+def generate_retry_policy(policy: RetryPolicy) -> urllib3.Retry:
```
Comment (der-gabe) [inline osctiny/utils/session.py:38]: I was wondering about the need for such a factory (and the custom `RetryPolicy` class to go along with it)... and then I read the version handling code and understood...

Well done on hiding that bit of complexity in its own neat little package! :+1:

File: osctiny/osc.py [hunk 3]
```diff
@@ -142,6 +142,8 @@ class Osc:
     default_timeout = (60, 300)
     default_connection_retries = 5
     default_retry_timeout = 5
+    retry_policy = RetryPolicy(max_attempts=default_connection_retries,
+                               backoff_max=default_retry_timeout)
```
Comment (der-gabe) [inline osctiny/osc.py:146]: Couldn't we just remove these class attributes altogether now?

**Comment by der-gabe** [comment]
Why was the request logging removed?

**Comment by der-gabe** [comment]
LGTM, otherwise! :slightly_smiling_face: :+1:

File: osctiny/osc.py (hunk 3 above)
Comment (crazyscientist) [reply osctiny/osc.py:146]: I would prefer not to break clients by simply removing the attributes, yet.

**Comment by crazyscientist** [comment]
> Why was the request logging removed?

By using `urllib3.Retry` the amount of logging gets reduced anyway and log messages might contain sensitive information.

Also, by removing it, we remove the potential for weird behavior (e.g. trying to write entire large files into the logs, ...)

File: osctiny/osc.py (hunk 3 above)
Comment (der-gabe) [reply osctiny/osc.py:146]: Oh, good point. Library users may instantiate `Osc` and set those two attributes... yeah, no, we shouldn't break that! :sweat_smile: 

Although it does raise the question how to deprecate and eventually get rid of them, but that's a future problem.

**Pull request merged**